src/mono/mono.proj

```diff
@@ -201,6 +202,23 @@
       <!-- Force running as arm64 even when invoked from an x86 msbuild process -->
       <_MonoBuildEnv Condition="'$(Platform)' == 'arm64'" Include="arch -arch arm64" />
     </ItemGroup>
+    <!-- Mac Catalyst specific options -->
+    <ItemGroup Condition="'$(TargetsMacCatalyst)' == 'true'">
+      <_MonoCMakeArgs Include="-DMAC_CATALYST=1" />
```

Comment:
the variable name is a bit too generic IMHO, I'd prefer something like `-DCMAKE_SYSTEM_VARIANT=maccatalyst` that we then use in our src/mono/CMakeLists.txt to set `HOST_MACCATALYST`